pulp_2to3_migration/app/migration.py

```diff
@@ -206,22 +187,32 @@
         pb.total = pulp2distributors_qs.count()
         pb.save()
 
-        pulp3_repo_setup = plan.get_pulp3_repository_setup()
-        if not pulp3_repo_setup:
-            for pulp2dist in pulp2distributors_qs:
-                dist_migrator = distributor_migrators.get(pulp2dist.pulp2_type_id)
-                await migrate_repo_distributor(pb, dist_migrator, pulp2dist)
-        else:
-            for repo_name in pulp3_repo_setup:
-                repo_versions_setup = pulp3_repo_setup[repo_name]['versions']
-                for repo_dist in repo_versions_setup:
-                    # find pulp2repo by id
-                    migrated_repo = Pulp2Repository.objects.get(pulp2_repo_id=repo_dist.repo_id)
-                    for dist_id in repo_dist.dist_ids:
-                        pulp2dist = Pulp2Distributor.objects.get(pulp2_object_id=dist_id)
-                        dist_migrator = distributor_migrators.get(pulp2dist.pulp2_type_id)
-                        await migrate_repo_distributor(
-                            pb, dist_migrator, pulp2dist, migrated_repo.pulp3_repository_version)
+        # gather all needed plugin distributor migrators
+        distributor_migrators = {}
+
+        for plugin in plan.get_plugin_plans():
+            if not plugin.migrator:
+                continue
+            distributor_migrators.update(**plugin.migrator.distributor_migrators)
+
+            pulp3_repo_setup = plugin.get_repo_creation_setup()
+            if not pulp3_repo_setup:
+                for pulp2dist in pulp2distributors_qs:
+                    dist_migrator = distributor_migrators.get(pulp2dist.pulp2_type_id)
+                    await migrate_repo_distributor(pb, dist_migrator, pulp2dist)
+            else:
+                for repo_name in pulp3_repo_setup:
+                    for repo_dist in pulp3_repo_setup[repo_name]['versions']:
+                        pass
+                        # find pulp2repo by id
```

Comment:
This code is broken, but I think it is also broken _currently_, I just never reach it on master branch because of https://pulp.plan.io/issues/5899

What currently gets returned by ```plan.get_pulp3_repo_setup()``` (both on master and in this PR) looks like [0] which isn't what this code expects I think.

[0]https://github.com/pulp/pulp-2to3-migration/pull/66/files#diff-2e8369e7a70124f8b4f1bc6f4120718eR171-R180 

I think it's supposed to be doing something more like this, correctt?

```
migrated_repo = Pulp2Repository.objects.get(pulp2_repo_id=repo_dist)
repo_distributors = Pulp2Distributor.objects.filter(pulp2_repository_id=migrated_repo.pk)
for pulp2dist in repo_distributors:
dist_migrator = distributor_migrators.get(pulp2dist.pulp2_type_id)
await migrate_repo_distributor(
     pb, dist_migrator, pulp2dist, migrated_repo.pulp3_repository_version)
```

Reply:
I think the code expects this https://pulp.plan.io/issues/5837#Example-of-parsed-plan.
I'd defer it to @ipanova since she worked on it.

Reply:
I think the code expects this https://pulp.plan.io/issues/5837#Example-of-parsed-plan.
I'd defer it to @ipanova since she worked on it.

Reply:
Eventually yes (I'm working on that issue) but right now it isn't getting data formatted like that.  I'm not sure if I should just leave it commented out for now (until that patch is finished) or if I should add a workaround here.

Reply:
I think @ipanova imlpied that it would work with the 5837 implemented only.
The workaround you suggested won't work in case distributor(s) should be used from a different repo. I think this is the main reason we need 5837.

Reply:
Then I'll just leave it commented out until 5837